cachecontrol/filewrapper.py

```diff
@@ -117,3 +117,9 @@ def _safe_read(self, amt: int) -> bytes:
             self._close()
 
         return data
+
+    def close(self):
```

Comment:
Not sure I follow this: there's a `_close` method too, and it seems to be disjoint in implementation from this one. Does this have an effect on the rest of the changes in this PR?

Reply:
FileWrapper is assigned as a `HTTPResponse._fp` which if it as a close method gets called by HTTPResponse.close` which if the request is streaming needs to be called by using the response as a context manager

Reply:
The windows issues are new and exciting, I'll have to investigate them later

Reply:
The `_close` is disjoint because it's called when the fp is exhausted and the buf has finished being written to, CallbackFileWrapper.close is called when the consumer of the response failed reading the body in some way, if the response was streaming

Reply:
> Does this have an effect on the rest of the changes in this PR?

It's needed otherwise you get a ResourceWarning that's propagated as an unraisable exception

Reply:
I resolved the windows issues - turns out there's a bug in cpython! I think I've found a cpython bug every time I've added filterwarnings=["error"] to a project!